client/go/outline/config/config_iptable_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/configyaml"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/stretchr/testify/require"
+)
+
+// mockSubDialerParser returns a mock parsing function for our tests.
+// This function simulates the behavior of a real sub-dialer parser.
+// It looks for a "name" in the config and returns the corresponding mock dialer.
+func mockSubDialerParser(dialers map[string]transport.StreamDialer) configyaml.ParseFunc[*Dialer[transport.StreamConn]] {
+	return func(ctx context.Context, config configyaml.ConfigNode) (*Dialer[transport.StreamConn], error) {
+		configMap, ok := config.(map[string]any)
+		if !ok {
+			return nil, errors.New("config is not a map[string]any")
+		}
+
+		name, ok := configMap["name"].(string)
+		if !ok {
+			return nil, errors.New("mock dialer config must have a 'name'")
+		}
+
+		dialer, ok := dialers[name]
+		if !ok {
+			return nil, fmt.Errorf("no mock dialer found with name: %s", name)
+		}
+
+		return &Dialer[transport.StreamConn]{
+			Dial: dialer.DialStream,
+		}, nil
+	}
+}
+
+func TestParseIPTableStreamDialer(t *testing.T) {
+	ctx := context.Background()
+
+	// Define a set of mock dialers for our tests to use.
+	mockDialers := map[string]transport.StreamDialer{
+		"dialerA": &mockStreamDialer{name: "dialerA"},
+		"dialerB": &mockStreamDialer{name: "dialerB"},
+		"default": &mockStreamDialer{name: "default"},
+	}
+
+	// Create a parser instance with our mock dialers.
+	parser := mockSubDialerParser(mockDialers)
+
+	// Define test cases
+	testCases := []struct {
+		name        string
+		config      map[string]any
+		expectErr   bool
+		expectedErr string
```

Comment:
*nit* these two can be combined. `""` `string` represents no error, otherwise compare it with the error message.

```suggestion
		expectErr   string
```